server/zio-http-server/src/test/scala/sttp/tapir/server/ziohttp/ZioHttpTestServerInterpreter.scala

```diff
@@ -21,15 +23,55 @@ class ZioHttpTestServerInterpreter(eventLoopGroup: EventLoopGroup, channelFactor
 
   override def server(routes: NonEmptyList[Http[Any, Throwable, Request, Response]]): Resource[IO, Port] = {
     implicit val r: Runtime[Any] = Runtime.default
-    val layers: ZLayer[Any, Nothing, EventLoopGroup with ServerChannelFactory] =
-      ZLayer.succeed(eventLoopGroup) ++ ZLayer.succeed(channelFactory)
 
-    val server: Server[Any, Throwable] = Server.app(routes.toList.reduce(_ ++ _))
+    val makeNettyDriver = {
+      import zio.http.netty.server._
+      import io.netty.bootstrap.ServerBootstrap
+      import io.netty.channel._
+      import io.netty.util.ResourceLeakDetector
+      import zio._
+      import zio.http.netty._
+      import zio.http.service.ServerTime
+      import zio.http.{Driver, Http, HttpApp, Server, ServerConfig}
 
-    val io: ZIO[Scope, Throwable, Server.Start] = ZIO
-      .scoped(Server.make(server ++ Server.port(0)))
-      .provide(layers)
+      import java.net.InetSocketAddress
+      import java.util.concurrent.atomic.AtomicReference
 
-    Resource.scoped[IO, Any, Int](io.map(_.port))
+      type ErrorCallbackRef = AtomicReference[Option[Server.ErrorCallback]]
```

Comment:
This got a bit messy, but more or less lifted out of NettyDriver.scala and allows us to share the underlying threadpools (which, as you said, greatly speeds up the tests). :)

Reply:
thanks! :)